scipy/stats/_sensitivity_analysis.py

```diff
@@ -0,0 +1,634 @@
+            Bootstrap result containing the confidence intervals and the
+            bootstrap distribution of the indices.
+
+            An object with attributes:
+
+            first_order : BootstrapResult
+                Bootstrap result of the first order indices.
+            total_order : BootstrapResult
+                Bootstrap result of the total order indices.
+            See `BootstrapResult` for more details.
+
+        """
+        def statistic(idx):
+            f_A_ = self._f_A[:, idx]
+            f_B_ = self._f_B[:, idx]
+
+            s, _ = self._f_A.shape
+            f_AB_ = self._f_AB.reshape((s, -1, n))
+            f_AB_ = f_AB_[:, :, idx].reshape(s, -1)
+
+            return self._indices_method(f_A_, f_B_, f_AB_)
+
+        n = self._f_A.shape[1]
+
+        res = bootstrap(
+            [np.arange(n)], statistic=statistic, method="BCa",
+            n_resamples=n_resamples,
+            confidence_level=confidence_level,
+            bootstrap_result=self._bootstrap_result
+        )
+        self._bootstrap_result = res
+
+        first_order = BootstrapResult(
+            confidence_interval=ConfidenceInterval(
+                res.confidence_interval.low[0], res.confidence_interval.high[0]
+            ),
+            bootstrap_distribution=res.bootstrap_distribution[0],
+            standard_error=res.standard_error[0],
+        )
+        total_order = BootstrapResult(
+            confidence_interval=ConfidenceInterval(
+                res.confidence_interval.low[1], res.confidence_interval.high[1]
+            ),
+            bootstrap_distribution=res.bootstrap_distribution[1],
+            standard_error=res.standard_error[1],
+        )
+
+        return BootstrapSobolResult(
+            first_order=first_order, total_order=total_order
+        )
+
+
+class PPFDist(Protocol):
+    @property
+    def ppf(self) -> Callable[..., float]:
+        ...
+
+
+def sobol_indices(
+    *,
+    func: Callable[[np.ndarray], npt.ArrayLike] |
+          Dict[Literal['f_A', 'f_B', 'f_AB'], np.ndarray],  # noqa
+    n: IntNumber,
+    dists: List[PPFDist] | None = None,
+    method: Callable | Literal['saltelli_2010'] = 'saltelli_2010',
+    random_state: SeedType = None
+) -> SobolResult:
+    r"""Global sensitivity indices of Sobol'.
+
+    Parameters
+    ----------
+    func : callable or dict(str, array_like)
+        If `func` is a callable, function to compute the Sobol' indices from.
+        It's signature must be::
+
+            func(x: ArrayLike) -> ArrayLike
+
+        with ``x`` of shape ``(d, n)`` and the output should have a shape
+        ``(s, n)`` with ``s`` the number of outputs and ``n`` the number of
+        samples.
+
+        If `func` is a dictionnary, contains the function evaluations from 3
+        different arrays. Keys must be: ``f_A``, ``f_B`` and ``f_AB``.
+        ``f_A`` and ``f_B`` should have a shape ``(s, n)`` and ``f_AB``
+        should have a shape ``(s, d*n)``.
```

Comment:
It is not clear to me here what the 3 different arrays refer to

Reply:
Thanks @j-bowhay, I tried to rephrase, let me know if this is better.

Reply:
Thanks that is much clearer now. Could you put a similar explanation as you have for `method` under description of `func` too (the description of the dictionary keys)?